napari/layers/vectors/vectors.py

```diff
@@ -242,9 +241,8 @@ def __init__(
 
         # Data containing vectors in the currently viewed slice
         self._view_data = np.empty((0, 2, 2))
-        self._displayed_stored = []
-        self._view_indices = []
-        self._view_alphas = []
+        self._view_indices: Union[List[int], np.ndarray] = []
```

Comment:
Given the other updates you made, you can simplify the type if the initial assignment is to an array:

```suggestion
        self._view_indices: np.ndarray = np.array([], dtype=int)
```